framework/src/modules/token/stores/supported_tokens.ts

```diff
@@ -150,24 +161,30 @@ export class SupportedTokensStore extends BaseStore<SupportedTokensStoreData> {
 
 	public async removeSupportForToken(context: StoreGetter, tokenID: Buffer): Promise<void> {
 		if (this._isMainchainOrNative(tokenID)) {
-			return;
+			throw new Error('All tokens are supported.');
```

Comment:
I think this error should go to the section below, where `allSupported` is being checked.

The LIP only covers the case for trying to remove LSK token, but it makes sense to handle LSK and native tokens as 1 case and throw error: _Can not remove support for LSK or a native token_.

Reply:
It was there initially until it was rightly pointed out that the private method `_isMainchainOrNative()` is already doing this check. Hence it's made more sense to throw the error where this private method is being invoked. WDYT?

Reply:
What I'm saying is that the error message here is wrong. We have 2 different cases which require 2 different error messages:

1. a user tries to remove LSK or a native token [which is what you're handling here]
1. a user tries to remove a token of a foreign chain, while the local chain supports all tokens [which is what this error message suggests]

When a user tries to remove LSK or a native token, I think it's incorrect to give error message that _"All tokens are supported."_. Because we don't know that before we check `allSupported()` method.

Also, LIP specifies error message _"Can not remove support for LSK token."_, which is not handled in your code. I propose that attempt to remove LSK or native token is handled with the same error message, because it is checked by the same function, but we can also separate that in 2 cases.